segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataStore.java

```diff
@@ -704,8 +704,16 @@ static SegmentInfo newSegment(String name, Collection<AttributeUpdate> attribute
                     .build();
         }
 
+        // createSegment in Metadata uses this method to get SegmentInfo
+        static SegmentInfo recoveredSegment(SegmentProperties segmentProperties) {
```

Comment:
Use this in your `DebugSegmentContainer` above.

Reply:
Done.